app/models/miq_server.rb

```diff
@@ -608,4 +608,8 @@ def server_timezone
   def tenant_identity
     User.super_admin
   end
+
+  def objects_in_settings_chain
+    zone ? zone.objects_in_settings_chain << self : [self]
```

Comment:
it should not change state since top parent always return [self], but as @imtayadeway suggested, it is better to use + if there is no intend to change state.